test/hooks/achievement-rule-checker.test.js

```diff
@@ -41,6 +41,80 @@ describe('\'achievement-rule-checker\' hook', () => {
     assert.deepEqual(result[0].amount, 1);
   });
 
+  it('grants achievements multiple times for different scopes', async () => {
+    await app.service('events').create({
+      'name': 'ScopeEvent',
+      'user_id': user_id,
+      'context': {
+        'user_id': user_id,
+        'context': {
+          'course_id': 42,
+        },
+      },
+    });
+
+    let achievements = await app.service('achievements').find({
+      query: {
+        user_id: user_id,
+        name: 'ScopeAchievement',
+      }
+    });
+
+    assert.equal(achievements.length, 1);
+
+    await app.service('events').create({
+      'name': 'ScopeEvent',
+      'user_id': user_id,
+      'context': {
+        'user_id': user_id,
+        'context': {
+          'course_id': 1337,
+        },
+      },
+    });
+
+    achievements = await app.service('achievements').find({
+      query: {
+        user_id: user_id,
+        name: 'ScopeAchievement',
+      }
+    });
+
+    assert.equal(achievements.length, 2);
+  });
+
+  it('works with the global scope', async () => {
+    await app.service('events').create({
+      'name': 'GlobalEvent',
+      'user_id': 'User 1',
+      'context': {
+        'user_id': 'User 1',
+      },
+    });
+    await app.service('events').create({
+      'name': 'GlobalEvent',
+      'user_id': 'User 1',
+      'context': {
+        'user_id': 'User 1',
+      },
+    });
+    await app.service('events').create({
+      'name': 'GlobalEvent',
+      'user_id': 'User 2',
+      'context': {
+        'user_id': 'User 2',
+      },
+    });
+
+    let achievements = await app.service('achievements').find({
+      query: {
+        name: 'GlobalAchievement',
+      }
+    });
+
+    assert.equal(achievements.length, 2);
```

Comment:
Actually, this is wrong. The scope is empty in the yaml, but we just assume user_id for it. Actually here, only User 1 should be granted the GlobalAchievement. What we tested for here, is ```scope: [user_id]``` or an empty scope field.